server/CVCMIServer.cpp

```diff
@@ -615,9 +625,24 @@ int main(int argc, char** argv)
 		//and return non-zero status so client can detect error
 		throw;
 	}
+#ifdef VCMI_ANDROID
+	AndroidVMHelper envHelper;
+	envHelper.callStaticVoidMethod(AndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "killServer");
+#endif
 	delete VLC;
 	VLC = nullptr;
 	CResourceHandler::clear();
 
   return 0;
 }
+
+#ifdef VCMI_ANDROID
+
+void CVCMIServer::create()
+{
+	boost::this_thread::sleep(boost::posix_time::milliseconds(1000));

```

Comment:
I'll need to check if that sleep is actually needed. It looks like it might be a leftover (because client doesn't really do anything worthwhile at this time -- after requesting server start, it just sleeps waiting for server response), but I can't check at the moment.